packages/core/src/entities/App.ts

```diff
@@ -1,60 +1,114 @@
-import Repo from "./Repo"
-import Entity from "./Entity"
-import { ConnectorInterface } from "../connections/ConnectorInterface"
+import Repo from './Repo'
+import Role from './Role'
+import Entity from './Entity'
+import {
+  AragonArtifact,
+  AppIntent,
+  Abi,
+  AragonEnvironment,
+  AragonManifest,
+} from '../types'
+import { ConnectorInterface } from '../connections/ConnectorInterface'
 
 // TODO: Implement all properties and methods from the API spec (https://github.com/aragon/plumbery/blob/master/docs/app.md).
-// [x] address 	String 	The address of the app proxy contract (never changes).
-// [x] appId 	String 	The appName but encoded.
-// [-] (need registry name prop) appName 	String 	The app ENS identifier. E.g. "token-manager.aragonpm.eth"
-// [-] (ipfs) author 	String 	App author, from the repository. E.g. "Aragon Association".
-// [ ] chainId 	String 	Chain ID for this app.
-// [ ] codeAddress 	String 	The address of the app contract (changes with every major version).
-// [ ] contentUri 	String 	The location of the app content. Empty for special apps like the kernel. E.g. "ipfs:QmdLEDDfi…"
-// [ ] (ipfs) contractPath 	String 	Path of the contract. E.g. "contracts/TokenManager.sol"
-// [ ] (ipfs) description 	String 	App description, from the repository. E.g. "Manage an organization’s token supply and distribution.".
-// [ ] (ipfs) htmlPath 	String 	The path of the app HTML page. Stays empty if the app doesn’t have a frontend. E.g. /index.html
+// [ ] chainId 	String 	Chain ID for this app. -> Is this the Aragon environment name?
 // [ ] (ipfs) htmlUrl 	String 	The HTTP URL of the app HTML page. Uses the IPFS HTTP provider. E.g. http://gateway.ipfs.io/ipfs/QmdLEDDfi…/index.html
 // [ ] (ipfs) contentUrl 	String 	The HTTP URL of the app content. Uses the IPFS HTTP provider. E.g. http://gateway.ipfs.io/ipfs/QmdLEDDfi…/
-// [ ] (ipfs) icons 	{ src: String, sizes: String }[] 	Array of icons for the app (follows the web app manifest icons format).
-// [x] name 	String 	Name of the app, from the repository. E.g. "Tokens".
-// [x] registryAddress 	String 	Address of the aragonPM registry for this app.
-// [-] (need name prop) registry 	String 	Name of the aragonPM registry for this app. E.g. "aragonpm.eth"
-// [ ] (ipfs) sourceUrl 	String 	URL of the app source code.
-// [x] version 	String 	The current version of the app.
-// [x] kernelAddress 	String 	The address of the kernel.
-// [x] isForwarder 	Boolean 	Whether the app can act as a forwarder.
+// [ ] registry 	String 	Name of the aragonPM registry for this app. E.g. "aragonpm.eth"
 // [ ] tags 	String[] 	Tags associated with the app.
```

Comment:
Not sure how to get the tags data 🤔

Where do we store that information?

Reply:
Ah yes sorry, this is something that is specific to the client: https://github.com/aragon/aragon/blob/b3d3613fc6df8f3d93ea619854dc7c8489501362/src/aragonjs-wrapper.js#L41-L56

Let’s remove it!